tests/test_modeling_tf_common.py

```diff
@@ -201,6 +202,41 @@ def test_saved_model_creation(self):
             saved_model_dir = os.path.join(tmpdirname, "saved_model", "1")
             self.assertTrue(os.path.exists(saved_model_dir))
 
+    def test_onnx_compliancy(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+        INTERNAL_OPS = [
+            "Assert",
+            "AssignVariableOp",
+            "EmptyTensorList",
+            "ReadVariableOp",
+            "ResourceGather",
+            "TruncatedNormal",
+            "VarHandleOp",
+            "VarIsInitializedOp",
+        ]
+
+        with open(os.path.join(".", "utils", "tf_ops", "onnx.json")) as f:
+            onnx_ops = json.load(f)["Keras2ONNX"]["opset12"]
+
+        for model_class in self.all_model_classes:
+            model_op_names = set()
+
+            with tf.Graph().as_default() as g:
+                model = model_class(config)
+                model(model.dummy_inputs)
```

Comment:
Again not familiar with TF way of working, but the actual inputs in PT do change quite a bit the actual traced graph.

That means that `use_cache` graph vs `first_pass` graph look quite different. Also setting variable batch_length where it can be fixed (for instance input_ids is necessarily [B, 1] for `use_cache` graph on decoder) can link to greatly different performance later down the road.

What I'm trying to say is that this test will probably check that the Ops used in TF are valid for some ONNX opset, **it does not by any means that check it can/will export the best production ready graph.**

And the real hot path in production is almost always, decoder-only with `use_cache` (even input_ids [1, 1]) within a generation loop (I don't think TF has the generation loop optimized yet.)

Reply:
Here we are not testing the graph, we are loading the entire list of operators, the graph here is not optimized. To give you an example, For BERT we load the > 5000 operators, while the optimised graph for inference is only around 1200 nodes. The role of this test is just to be sure to have the entire list of used operators inside the list proposed here https://github.com/onnx/tensorflow-onnx/blob/master/support_status.md

Reply:
Yes I know, I was just emphasizing it.
Also `optimized` graph of ONNX can only go so far. It cannot know about `past_values` if they are not passed within those dummy inputs. 

unoptimized small graph > optimized big graph

- big as in, large sequences length, not sheer node number
- Again talking about PT here, I didn't check with TF yet, but results are probably similar.

Reply:
I think there is a misunderstanding, this test is only here to say "this TF op is also implemented in ONNX" nothing more. And not for testing if the optimized ONNX graph will work as expected or not.

Reply:
> I think there is a misunderstanding, this test is only here to say "this TF op is also implemented in ONNX" nothing more. 
There is no misunderstanding, I was trying to say  what you just said.

Reply:
Ok, so if you are trying to say the same thing there is no problem then^^